### ISSUE TITLE
Signal 11 caught. Segfault

### ISSUE THREAD

**Issue by srounet** [TuRz4m/Ark-docker#4]
Hi TuRz4m,

I'm trying to bootstrap a new server and can't get it to start.
Here is the arkmanager-user.cfg:

``` cfg
# --- USER CONFIG --- #
# ARK server options - use ark_<optionname>=<value>
# comment out these values if you want to define them
# inside your GameUserSettings.ini file
serverMap=${SERVERMAP}                                          # server map (default TheIsland)
serverMapModId=504122600
ark_SessionName=${SESSIONNAME}                                  # if your session name needs special characters please use the .ini instead
ark_ServerPassword=${SERVERPASSWORD}                            # ARK server password, empty: no password required to login
ark_ServerAdminPassword=${ADMINPASSWORD}                        # ARK server admin password, KEEP IT SAFE!
ark_MaxPlayers=${NBPLAYERS}                                      # Number MAX of player
# ark_bRawSockets=""                                            # Uncomment if you want to use ark raw socket (instead of steam p2p) [Not recommended]
arkflag_log=""


# ----- Mods ----- #
#ark_GameModIds="487516323,487516324,487516325"                     # Uncomment to specify additional mods by Mod Id separated by commas
ark_GameModIds="479295136"
# Mod OS Selection
mod_branch=Windows
# Add mod-specific OS selection below:
#mod_branch_496735411=Windows
# ----------------#

# ARK server flags - use arkflag_<optionname>=true
#arkflag_OnlyAdminRejoinAsSpectator=true                            # Uncomment to only allow admins to rejoin as spectator
#arkflag_DisableDeathSpectator=true                                 # Uncomment to disable players from becoming spectators when they die

# ARK server options - i.e. for -optname=val, use arkopt_optname=val
#arkopt_StructureDestructionTag=DestroySwampSnowStructures


# Update warning messages
# Modify as desired, putting the %d replacement operator where the number belongs
msgWarnUpdateMinutes="This ARK server will shutdown for an update in %d minutes"
msgWarnUpdateSeconds="This ARK server will shutdown for an update in %d seconds"
arkBackupPreUpdate="false"                                          # set this to true if you want to perform a backup before updating
arkwarnminutes="30"                                                 # number of minutes to warn players when using update --warn
```

arkserver.log

``` cfg
00:19:44: start
00:19:44: Running /ark/server/ShooterGame/Binaries/Linux/ShooterGameServer -MapModID=504122600\?GameModIds=479295136\?MaxPlayers=70\?Port=7777\?QueryPort=27015\?RCONEnabled=True\?RCONPort=32330\?ServerAdminPassword=xxxx\?ServerPassword\?SessionName=\[FR\]\ Krokmou\ Valhalla\?listen -log
00:19:45: Server PID: 3046
[S_API FAIL] SteamAPI_Init() failed; SteamAPI_IsSteamRunning() failed.
Setting breakpad minidump AppID = 346110
Signal 11 caught.
/usr/local/bin/arkmanager: line 558:  3046 Segmentation fault      "$arkserverroot/$arkserverexec" "$arkserveropts" "${arkextraopts[@]}"
00:20:32: Bad PID ''; expected '3046'
00:20:32: exited with status 0
```

update.log

``` cfg
00:18:51: Mod 479295136 updated
00:19:43: Mod 504122600 updated
```

The server should run Valhalla map and Aku Shima mod.
If you have time, can you confirm that the server either crash or won't start using this configuration ?

I can't figure out what to look, both mods are downloaded correctly (it seems) and Ark is successfully installed.
Thank you.

**Comment by TuRz4m** [comment]
Hi,

I just launch an instance with :
- your arkmanager.cfg in /data/arktest
- the rest of the directory is empty 
- the commandline : `docker run -it -p 7779:7779 -p 7779:7779/udp -p 27016:27016 -p 27016:27016/udp -p 32331:32330 -e SESSIONNAME="Docker-Ark - TEST" -e ADMINPASSWORD="pass" -e NBPLAYERS=10 -e SERVERPASSWORD="pass" -e STEAMPORT=7779 -e SERVERPORT=27016 -v /data/arktest:/ark --name arktest turzam/ark`  

This doesn't work. I have a segfault too.
**But** if I change the SESSIONNAME to "test", it works -_-

So... after more testing... 
SESSIONNAME='this is a test' : segfault
SESSIONNAME='aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa' : it works
SESSIONNAME="this is a test" : segfault
SESSIONNAME="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" : it works

In arkmanager.cfg I changed the line `ark_SessionName=${SESSIONNAME}` to `ark_SessionName="${SESSIONNAME}"` 
It still doesn't work with `SESSIONNAME="this is a test"`

Naturally, I also tested with `ark_SessionName="this is a test"`... segfault...

And to finish... I tried   `SESSIONNAME="this is a test"` without the `serverMapModId=504122600` and it works.

TL;DR : I don't know why it doesn't work but it's somehow link to the spaces in the session name.

**Comment by srounet** [comment]
With the same configuration as before, I'm able to launch the serveur with a custom name (this is what makes it segfault).

I commented `#ark_SessionName=${SESSIONNAME}` and used the GameUserSettings.ini to change the name and it worked.
As a rule of thumb, I'm not using `ark_SessionName` anymore from now.
